Basic functionality for user

VladimirSlavik opened the pull request:
Just user name, gecos, and defaults.

poncovka committed:
Add the set_repository_enabled function

Use the function to enable or disable a repository.

VladimirSlavik committed:
Simplify release notes (#infra)

Also, make sure there is no newline at end.

VladimirSlavik committed:
Basic functionality for user

KKoukiou commented on ui/webui/src/users/UserAccount.jsx:
```diff
@@ -35,7 +84,14 @@ export const UserAccount = () => {
-                Not implemented
+                <Form isHorizontal>
+                    <FormGroup label='Real name'>
+                        <TextInput id='gecos' onChange={onGecosChange} placeholder='John Doe' />
```
You can just directly use setGecos state setter function instead of the wrapper that just calls this ;)

KKoukiou commented on ui/webui/src/users/UserAccount.jsx:
```diff
@@ -35,7 +84,14 @@ export const UserAccount = () => {
+                    <FormGroup label='Real name'>
+                        <TextInput id='gecos' onChange={onGecosChange} placeholder='John Doe' />
+                    </FormGroup>
+                    <FormGroup label='User name'>
```
`FormGroup` component has a required property called 'fieldId'. This should be the `id` of the field that is nested in it, in this case 'name'. The reason for this is accessibility of the html code, https://developer.mozilla.org/en-US/docs/Learn/Accessibility/What_is_accessibility

How this increases accessibility?
It adds an `for` attribute to the DOM label element, making the `label` point to the `input`. This has also another effect, that when the user will then click on the label, the input box will get focused, which is the desired behavior.

If you want to know more let me know and I can direct you to the correct documentation.

KKoukiou commented on ui/webui/src/users/UserAccount.jsx:
```diff
@@ -15,16 +15,65 @@
+        setGecos(value);
+    };
+
+    useEvent(usersProxy, 'changed', (event, data) => {
```
The contents of this are called in the following times:
* when the proxy is ready and it's data are ready for reading
* every time some property changes

You are mostly interested in the first bullet, so when the proxy data become available for the first time.

Then you want to initialize the UI with possibly existing users (set from kickstart)

For that it's probably enough to do here:

``` js
if (data.Users.length > 0)
    setName(data.Users[0]).name.v;
```

Similarly for the real name.

KKoukiou reviewed (commented):
That's a really great start! I left some comments, tried to be verbose, I promise I won't be like this forever :D

VladimirSlavik committed:
Run chown instead of os.walk-ing to re-own home dir

- Provides the same result,
- chown is older and widely used so bugs are less likely,
- bugs are not in our code,
- Python stack exhaustion no longer happens.

Improve also docstrings and comments.

Resolves: rhbz#2036415

KKoukiou committed:
webui: tests: add info on how to run these in a toolbox (#docs)

KKoukiou committed:
webui: parameterize ports for ssh, cockpit connection and http server

Previously no parallel test runs were possible, since there was port collision on the host.
This commit makes it possible for local test runs or CI to run in
paralle.

M4rtinK committed:
Merge pull request #3778 from M4rtinK/master-cockpit_web_ui_prototype

Inital Web UI bootstrapping

KKoukiou committed:
webui: start translating some strings

Utilize cockpit's localization mechanism for making this happen [1].

This commit also disables eslint's quotes rules as these force single
quotes by default. However, cockpit's translation mechanism works only
with double quoted strings.

So, this commit just uses the cockpit's gettext like API for translating strings.
Anaconda's existing pot file generation (see po directory) needs to be
extended to find translatable strings in this directory.

[1] https://cockpit-project.org/guide/latest/cockpit-locale

KKoukiou committed:
webui: eslint: let's always prefer double quotes since cockpit localization needs it

rvykydal committed:
Use systemd-resolved in installer environment.

Following https://fedoraproject.org/wiki/Changes/systemd-resolved

VladimirSlavik committed:
Merge pull request #3781 from VladimirSlavik/master-chown-homedir

Run chown instead of os.walk-ing to re-own home dir

VladimirSlavik committed:
Merge pull request #3776 from VladimirSlavik/master-release-changelog-heading

Simplify release notes (#infra)

KKoukiou committed:
po: start translating webui files (jsx) and add support for cockpit translation functions

Let's create seperate pot file for the jsx suffixed files, as without
--language=JavaScript this gives:

xgettext: warning: file 'ui/webui/src/timezone/TimeDate.jsx' extension 'jsx' is unknown; will try C
...

For letting webui code have access to translations, symlink ui/webui/po
directory, to let webpack pick up the po files when
bundling the code.

KKoukiou committed:
po: add hack to workaround semicolon bug

HACK: Semicolon in some languages in 'Plural-Forms' header key can break translations
https://github.com/mikeedwards/po2json/issues/87
Removing all semicolons is safe way around this.

KKoukiou committed:
webui: po: cockpit-po-plugin expects that translations are from the current directory

Instruct it otherwise.

poncovka committed:
Merge pull request #3780 from poncovka/master-repo_enabled

Add the set_repository_enabled function

poncovka committed:
Add the `enabled` attribute to the repository configuration data

Extend the DBus data for the repository configuration with the `enabled`
property.

KKoukiou committed:
po: stop using --use-fist when creating the anaconda.pot file

This option causes the resulting pot file not list all the files
containing a translatable string, resulting in the webui po parsing code
to not being able to detect if a translation is from the webui code or
not.

KKoukiou committed:
po: don't translate the externally fetched pkg/lib content for now

If we end up using any of these files containing translatable
strings we should maybe change approach, but right now, this just
creates noise.

rvykydal committed:
Merge pull request #3783 from rvykydal/use-systemd-resolved-in-installer

Use systemd-resolved in installer environment.

KKoukiou committed:
spec: list webui language translation files into the RPM

KKoukiou committed:
Merge pull request #3782 from KKoukiou/toolbox-document

webui: tests: add info on how to run these in a toolbox

poncovka committed:
Test a recreation of the same thread

Add a test for recreation of the same thread. There is an unresolved bug
in Anaconda, so the test is marked as an expected failure.

KKoukiou committed:
build: Remove make as part of run-build-and-arg script

Otherwise re-building webui as part of the container-rpm-test
skips running webpack, resulting in translations js files to not get generated.

Add po-pull target as a dependency to the scratch target.

Translation files should be part of the RPM and there are currently
checks for this in the spec file (only for the webui translations)

Make sure to download translations before building the RPMs, for the
RPM unit checks to pass.

KKoukiou committed:
Merge pull request #3790 from KKoukiou/test-parallel-runs

webui: parameterize ports for ssh, cockpit connection and http server

KKoukiou committed:
packit: release: unset use_cockpit by sedding the specfile in packit script

Also, remove the unpackaged anaconda-webui files in the case `use_cockpit` is unset.

KKoukiou committed:
Merge pull request #3794 from KKoukiou/use_cockpit

packit: release: unset use_cockpit by sedding the specfile in packit script

M4rtinK committed:
New version - 36.16

KKoukiou committed:
Merge pull request #3785 from KKoukiou/translations-2

po: start translating webui files (jsx) and add support for cockpit translation functions

poncovka committed:
Merge pull request #3791 from poncovka/master-thread_wait

Test a recreation of the same thread

poncovka committed:
Clean up the code for including Web UI in an updates image

The makeupdates script should check if ui/webui/dist exists and copy it only
once. Otherwise, the script will fail or copy the same directory multiple times.

poncovka committed:
Merge pull request #3787 from poncovka/master-extend_repo_data

Add the `enabled` attribute to the repository configuration data

poncovka committed:
Merge pull request #3795 from poncovka/master-fix_webui_updates

Clean up the code for including Web UI in an updates image
